time_span/time_span.go

```diff
@@ -2,28 +2,210 @@ package time_span
 
 import "time"
 
-// A closed interval of time.
-type TimeSpan struct {
+// Return the earlier of the two times
+func MinTime(a time.Time, b time.Time) time.Time {
+	if a.Before(b) {
+		return a
+	} else {
+		return b
+	}
+}
+
+// Return the kater of the two times
+func MaxTime(a time.Time, b time.Time) time.Time {
+	if a.Before(b) {
+		return b
+	} else {
+		return a
+	}
+}
+
+type BaseInterval struct {
 	Start time.Time `json:"start"`
 	End   time.Time `json:"end"`
 }
 
-func NewTimeSpan(start time.Time, end time.Time) *TimeSpan {
+// A closed interval of time.
+type ClosedInterval BaseInterval
+
+func NewClosedInterval(start time.Time, end time.Time) ClosedInterval {
 	if start.After(end) {
 		start, end = end, start
 	}
 
-	return &TimeSpan{
+	return ClosedInterval{
 		Start: start,
 		End:   end,
 	}
 }
 
-func (span TimeSpan) Duration() time.Duration {
-	return span.End.Sub(span.Start)
+func (span ClosedInterval) Duration() time.Duration {
+	if span.End.After(span.Start) {
+		return span.End.Sub(span.Start)
+	} else {
+		return time.Duration(0)
+	}
```

Comment:
Having no away to represent empty intervals also seems surprising.

If Includes does not work for empty intervals, I'll fix it, but I'd rather that Intersect returns empty intervals than that it return an error.

Reply:
If we're letting `Start > End` represent the empty intersection, that's fine, but let's document this. Rather than carefully thinking about whether our operations work for the cases where `Start > End`, what if we introduced an `IsEmpty()` and checked this case explicitly? I still think that `Expand` needs to be able to return an error, though.